src/Interpreters/Cache/QueryCache.cpp

```diff
@@ -129,6 +138,7 @@ QueryCache::Key::Key(
     , is_shared(is_shared_)
     , expires_at(expires_at_)
     , is_compressed(is_compressed_)
+    , query_string(queryStringFromAst(ast_))
```

Comment:
can there be a huge amount of QueryCache keys? is it ok to add query string for each in terms of memory? just in case user has a lot of really big queries

Reply:
I guess this is only a theoretical problem.
1. The cache is supposed to be used only with *few* expensive queries.
2. The default upper limit on the number of cache items is 1024 ([query_cache.max_entries](https://clickhouse.com/docs/en/operations/server-configuration-parameters/settings#server_configuration_parameters_query-cache)).
3. The query strings would have to be super long to occupy significant space (close to [max_query_size](https://clickhouse.com/docs/en/operations/settings/settings#settings-max_query_size)).

Following from 2. and 3., the absolute worst case space consumption for the query strings is 262 MB. Not great but also not a value that would break the server.